presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```diff
@@ -142,4 +133,116 @@ public int size()
             }
         };
     }
+
+    private class KeySet
```

Comment:
How are those broken, other than returning a read-only or inefficient collection? But I agree, if we don't need them, we should probably just make them throw.